### ISSUE TITLE
null terminated string (Pinvoke)

### ISSUE THREAD

**Issue by Bxaa** [dotnet/runtimelab#1741]
```
<DllImport("user32.dll", EntryPoint:="EnumDisplayDevices", CharSet:=CharSet.Auto, CallingConvention:=CallingConvention.StdCall)>
        Public Shared Function EnumDisplayDevices(lpDevice As String, iDevNum As UInteger, ByRef lpDisplayDevice As DISPLAY_DEVICE, dwFlags As UInteger) As Integer
        End Function
Dim c_Dev As New DISPLAY_DEVICE With {.cb = Marshal.SizeOf(c_Dev)}
EnumDisplayDevices(Nothing, 0, c_Dev, 0)
Dim p_Adapter As String = c_Dev.DeviceString

```
NET 5 return correct string (CharSet:=CharSet.Auto)
AOT return string + Chr(0),Chr(0),Chr(0),Chr(0),Chr(0),.........(CharSet:=CharSet.Auto)
Any other CharSet return nothing (AOT)

All day long I couldn't figure out what was wrong...:))
I noticed it only in the log file with enabled all system chars

Marshal issue?
Perhaps the problem is deeper and not just here
Yep, .TrimEnd(Chr(0), Chr(32)) fix issue...

**Comment by jkotas** [comment]
The code fragment that you have shared references  `DISPLAY_DEVICE` type. The problem is likely in how NativeAOT treats the `DISPLAY_DEVICE` type. Could you please share a code fragment that compiles (e.g. including `DISPLAY_DEVICE` type)?

**Comment by Bxaa** [comment]
https://drive.google.com/file/d/1Gieu7aoMWO_p-wi7tvTse8iADhGkZdJl/view?usp=sharing

Thx for your work

**Comment by MichalStrehovsky** [comment]
Fixed in #1743. Updated ILCompiler nuget will be out within 24 hours.

**Comment by Bxaa** [comment]
Thx!
